src/Sign.Cli/CertificateStoreCommand.cs

```diff
@@ -84,16 +87,38 @@ internal CertificateStoreCommand(CodeCommand codeCommand, IServiceProviderFactor
                     return;
                 }
 
-                // SHA-1 Thumbprint is required in case the provided certificate container contains multiple certificates.
-                if (string.IsNullOrEmpty(sha1Thumbprint))
+                // Certificate Fingerprint is required in case the provided certificate container contains multiple certificates.
+                if (string.IsNullOrEmpty(certificateFingerprint))
                 {
                     context.Console.Error.WriteLine(
-                        FormatMessage(Resources.InvalidSha1ThumbprintValue, Sha1ThumbprintOption));
+                        FormatMessage(Resources.InvalidCertificateFingerprintValue, CertificateFingerprintOption));
                     context.ExitCode = ExitCode.NoInputsFound;
 
                     return;
                 }
 
+                HashAlgorithmName certificateFingerprintAlgorithm = HashAlgorithmName.SHA256;
+
+                // CertificateFingerprintAlgorithm defaults to SHA-256, but one could pass in an empty argument through PowerShell scripts.
+                if (unparsedCertificateFingerprintAlgorithm != null)
+                {
+                    // Remove "-" from the unparsed entry to match with HashAlgorithmName.
+                    unparsedCertificateFingerprintAlgorithm = string.Join(string.Empty, unparsedCertificateFingerprintAlgorithm.Split('-'));
+
+                    // We only accept SHA256, SHA384, and SHA512 as valid fingerprint algorithms.
+                    if (!Enum.TryParse<HashAlgorithmName>(unparsedCertificateFingerprintAlgorithm, ignoreCase: true, out certificateFingerprintAlgorithm)
```

Comment:
`System.Security.Cryptography.HashAlgorithmName` is a struct not an enum, so `Enum.TryParse<HashAlgorithmName>(...)` will throw a `System.ArgumentException` exception.  This code doesn't work.